lighthouse-core/audits/image-aspect-ratio.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+ /**
+  * @fileoverview Checks to see if the aspect ratio of the images used on
+  *   the page are equal to the aspect ratio of their display sizes. The
+  *   audit will list all images that don't match with their display size
+  *   aspect ratio.
+  */
+'use strict';
+
+const Audit = require('./audit');
+
+const URL = require('../lib/url-shim');
+
+class ImageAspectRatio extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Images',
+      name: 'image-aspect-ratio',
+      description: 'Uses Images with appropriate aspect ratio',
+      failureDescription: 'Does not uses Images with appropriate aspect ratio',
+      helpText: 'Serve Images that are appropriately sized as per the aspect ratio',
+      requiredArtifacts: ['ImageUsage']
+    };
+  }
+
+  /**
+   * @param {!Object} image
+   * @return {?Object}
+   */
+  static computeAspectRatios(image) {
+    const url = URL.elideDataURI(image.src);
+    const actualAspectRatio = (image.naturalWidth / image.naturalHeight).toFixed(2);
+    const usedAspectRatio = (image.clientWidth / image.clientHeight).toFixed(2);
+    const doesRatiosMatch = Math.abs(
+      parseFloat(actualAspectRatio) - parseFloat(usedAspectRatio)
+    ) < 0.5;
```

Comment:
0.5 is a bit large I meant .05 if you got this from my suggestion :) otherwise a 500x1000 would pass even if the original were 1000x1000, let's also keep this allowable slack consistent by checking the inverse ratio if they're both larger than 1

i.e. `if (usedAspectRatio > 1 && actualAspectRatio > 1) doRatiosMatch = Math.abs((1 / usedAspectRatio) - (1 / actualAspectRatio))...`